### ISSUE TITLE
Add policy/value network scaffold

### ISSUE THREAD

**Issue by ryparikh** [ryparikh/AdvanceWarsServer#7]
Part of the AlphaZero-style self-play training loop work.

## Context

State and action encodings now exist. The project needs a small, real policy/value model scaffold that can predict dense policy logits aligned to encoded actions and a scalar value from the current player's perspective, while staying modest enough to verify locally.

The scaffold should be useful to later training (#8) and neural-guided MCTS (#166), but should not implement training losses, replay loading, PUCT, or self-play integration itself.

Relevant code/docs:
- `AdvanceWarsServer/main.cpp` current LibTorch/MNIST experiment
- `AdvanceWarsServer/inc/StateTensor.h` and `AdvanceWarsServer/src/Tensor.cpp`
- `AdvanceWarsServer/inc/ActionSpace.h` and `AdvanceWarsServer/src/ActionSpace.cpp`
- `docs/STATE_TENSOR.md`
- `docs/ACTION_SPACE.md`
- `docs/TRAINING_DESIGN.md`

## Design Decisions

- Use `standard-gl-v1-state` input as batched `float32` `NCHW`: `[N, StateTensor::ChannelCount(), 23, 27]`.
- Use a small configurable residual CNN trunk by default: `hiddenChannels=64`, `residualBlocks=4`, `normGroups=8`.
- Use GroupNorm in residual blocks. Validate `hiddenChannels % normGroups == 0`.
- Initialize weights deterministically from a non-negative seed, default `0`: Kaiming/He for trunk/residual convs, zero biases, and small random final heads.
- Output raw dense policy logits with shape `[N, ActionSpace::ActionCount()]`.
- Implement policy as source-cell action planes plus global logits: `policyPlaneCount=2503`, `globalActionCount=3`, and validate `2503 * 23 * 27 + 3 == ActionSpace::ActionCount()`.
- Output scalar `tanh` value with shape `[N, 1]`, where `1` is current-player win, `0` is draw/neutral, and `-1` is current-player loss.
- Keep legal-action filtering/masking outside the model. #8 owns training-time masks/losses; #166 owns MCTS-time legal prior normalization.
- Expose a raw `torch::Tensor` forward path for future hot paths and a safe vector-based wrapper for encoded `std::vector<float>` batches.
- Require raw tensor inputs to be positive-batch, contiguous, `float32`, and exact `NCHW` shape. Return contiguous outputs.
- Inference helpers should use eval/no-grad. Checkpoint load should return an eval-mode model by default.
- Save checkpoint weights on CPU for portability; load can move to the requested device.
- Support device selection `cpu|auto|cuda`, default `cpu`. Explicit `cuda` fails clearly if unavailable.
- Keep the MNIST LibTorch experiment as an opt-in diagnostic behind `-torchlib --mnist-path <path>`, not as the project model path.

## Checkpoint Bundle

Use an inspectable checkpoint directory:

```text
checkpoint-dir/
  metadata.json
  model.pt
```

`metadata.json` should be pretty-printed ordered JSON with strict known fields only. It should include:

- `metadataVersion: 1`
- `modelVersion: "standard-gl-policy-value-v1"`
- `createdAt`
- `validatedDevice` as provenance only, not compatibility
- `seed` as a JSON number in `0..INT64_MAX`
- architecture settings: `hiddenChannels`, `residualBlocks`, `normGroups`, `policyPlaneCount`, `globalActionCount`
- state tensor metadata: version, channels, height, width
- action-space metadata: version and action count
- output metadata: policy logits count and value size
- `parameterCount` as provenance only

Checkpoint load should fail clearly if `metadata.json` or `model.pt` is missing, if unknown top-level fields exist, or if compatibility fields differ from the current model/state/action contract.

## Commands And Tests

- Add `-test-model` as the focused scaffold test command.
- Add `-model-init --out <checkpoint-dir> [--device cpu|auto|cuda] [--hidden-channels n] [--res-blocks n] [--norm-groups n] [--seed n] [--force]`.
- `-model-init` should create missing parent directories, refuse to overwrite an existing checkpoint directory by default, and with `--force` overwrite only known bundle files while rejecting unexpected files.
- `-model-init` should write the checkpoint, reload it on CPU, and run a small validation. If CUDA/auto selects CUDA, also validate requested-device inference.
- `-test-model` should cover default and tiny configs, synthetic batch inputs, a real Standard game encoded through `StateTensor::Encode`, output shapes, finite outputs, value range, policy-layout invariants, deterministic same-seed initialization, different-seed output change, CPU save/load output equivalence, checkpoint portability to CUDA when requested, and focused invalid-checkpoint cases.
- Generate temporary checkpoint artifacts for tests and clean them up on success. Do not check in model artifacts.

## Build And Docs

- Move LibTorch usage out of `main.cpp` into model/dev-test source files.
- Read LibTorch paths from configuration such as `LIBTORCH_ROOT` instead of hardcoded local paths.
- Support CPU-only LibTorch by default. Link CUDA-specific LibTorch/CUDA/cuDNN libraries only behind an explicit MSBuild property such as `UseLibtorchCuda=true`.
- Document `LIBTORCH_ROOT`, optional CUDA build configuration, required runtime `PATH`, `-test-model`, `-model-init`, checkpoint bundle layout, device options, and the preserved MNIST diagnostic path.
- Add `.gitignore` coverage for local checkpoint/model artifacts.
- Update `TRAINING_LOOP_WORK_ITEMS.md` only for scaffold-complete items; leave replay training, training losses, MCTS inference batching, and PUCT integration open.

## Explicit Deferrals

- #8 owns replay parsing, model-ready training batches, policy/value losses, optimizer state, and training checkpoints.
- #166 owns neural-guided MCTS, PUCT, legal prior normalization, and model inference batching inside search.
- #177 owns production architecture scaling, mixed precision/fp16, profiling, and richer architecture diagnostics.
- #178 owns async logging/checkpoint I/O if disk throughput becomes a bottleneck.

## Done When

The project can instantiate the Standard GL policy/value model, run CPU-default inference for batches of encoded states, optionally validate CUDA inference when configured, create and reload portable checkpoint bundles with strict metadata compatibility checks, and run `-test-model` successfully.
